### PR TITLE
[DataFrame] Implements df.pipe

### PR DESCRIPTION
**Pull request by SaladRaider**
## What do these changes do?
Implements df.pipe

### CODE REVIEW

File: python/ray/dataframe/test/test_dataframe.py [hunk 1]
```diff
@@ -2166,8 +2166,23 @@ def test_pct_change():
 def test_pipe():
```
Comment (kunalgosar) [inline python/ray/dataframe/test/test_dataframe.py]: Can we continue to use the pytest fixtures here?

**Review by kunalgosar — CHANGES_REQUESTED** [review]
Looks really good! Just a small comment regarding testing.

**Review by devin-petersohn — APPROVED** [review]

**Review by kunalgosar — APPROVED** [review]
Looks great!

**Comment by devin-petersohn** [comment]
Passes private-travis. Merged, Thanks @SaladRaider!

**Pull request merged**